include/tenstorrent/tt_boot_fs.h

```diff
@@ -74,21 +69,44 @@ typedef enum {
 	TT_BOOT_FS_CHK_FAIL,
 } tt_checksum_res_t;
 
-extern tt_boot_fs boot_fs_data;
-
 uint32_t tt_boot_fs_next(uint32_t prev);
 
-int tt_boot_fs_mount(tt_boot_fs *tt_boot_fs, tt_boot_fs_read hal_read, tt_boot_fs_write hal_write,
-		     tt_boot_fs_erase hal_erase);
+uint32_t tt_boot_fs_cksum(uint32_t cksum, const uint8_t *data, size_t size);
+
+/**
+ * @brief Read data from the boot filesystem at a specified address
+ *
+ * @param dev Flash device to read from
+ * @param addr Flash address to read from
+ * @param buffer Output buffer to store read data
+ * @param size Number of bytes to read
+ */
+int tt_bootfs_ng_read(const struct device *dev, uint32_t addr, uint8_t *buffer, size_t size);
+
+int tt_bootfs_ng_write(const struct device *dev, uint32_t addr, uint8_t *buffer, size_t size);
```

Comment:
I'm not entirely sure if these are useful currently, since it looks like they just duplicate the flash read and flash write apis.